File: src/main/java/ph/edu/cksc/college/advweb/blog/service/ProductService.java
package ph.edu.cksc.college.advweb.blog.service;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.web.multipart.MultipartFile;
import ph.edu.cksc.college.advweb.blog.model.Product;

import java.io.IOException;
import java.util.List;

public interface ProductService {
    Product createProduct(Product product);

    Product updateProduct(Product product, MultipartFile picture) throws IOException;

    Page<Product> getProducts(String query, Pageable pageable);

    Product getProductById(long productId);

    void deleteProduct(long id);
}

File: react-client/src/services/ProductService.js
import httpClient from "./HttpCommon";
import authHeader from "./AuthHeader";

export const listProducts = (params) => {
    return httpClient.get('/products', { params })
};

export const createProduct = (product) => {
    return httpClient.post('/products', product, { headers: authHeader() });
}

export const getProductById = (productId) => {
    return httpClient.get('/products/' + productId);
}

export const updateProduct = (productId, product) => {
    return httpClient.post('/products/' + productId, product, {
        headers: {
            "Content-Type": "multipart/form-data", ...authHeader()
        }
    });
}

export const deleteProduct = (productId) => {
    return httpClient.delete('/products/' + productId, { headers: authHeader() });
}